booking.py

```diff
@@ -145,36 +157,32 @@ def __set_selections(self, selections):
                    'TypeError: not a Selection object.')
         self.selections = selections
 
-    def __set_actions(self, actions):
+    def __set_actions(self, actions, variation):
         if not isinstance(actions, list):
             raise TypeError(
                     'TypeError: not a list object.')
         for action in actions:
             if not isinstance(action, Action):
                 raise TypeError(
                    'TypeError: not an Action object.')
-        self.actions = [self.__set_new_action(action) \
+        self.actions = [self.__set_new_action(action, variation) \
                 for action in actions]
 
-    def __set_new_action(self, action):
-        name = '#'.join([action.variable,
-            self.dataset.name,
-            '-'.join([selection.name for selection in self.selections])])
+    def __set_new_action(self, action, variation):
+        if variation is None:
+            name = '#'.join([self.dataset.name,
+                '-'.join([selection.name for selection in self.selections]),
+                'Nominal', action.name])
+        else:
+            if not isinstance(variation, Variation):
+                raise TypeError(
+                   'TypeError: not a Variation object.')
```

Comment:
You don't have to write `'Type Error: ...'`, because the exception will show this by itself. I think I've seen this somewhere a second time.

Reply:
oh yes, didn't notice. I'll add a commit where I fix all of them